mumd/src/state.rs

```diff
@@ -129,19 +132,21 @@ impl State {
             // this is someone else
             // send notification only if we've passed the connecting phase
             if matches!(*self.phase_receiver().borrow(), StatePhase::Connected(_)) {
-                let channel_id = msg.get_channel_id();
+                let this_channel = msg.get_channel_id();
+                let other_channel = self.get_users_channel(self.server().unwrap().session_id().unwrap());
+                //TODO can this fail?
+                let this_channel_name = self.server().unwrap().channels().get(&this_channel).map(|c| c.name());
 
-                if channel_id
-                    == self.get_users_channel(self.server().unwrap().session_id().unwrap())
-                {
-                    if let Some(channel) = self.server().unwrap().channels().get(&channel_id) {
+                if this_channel == other_channel {
+                    if let Some(this_channel_name) = this_channel_name {
                         notifications::send(format!(
                             "{} connected and joined {}",
-                            &msg.get_name(),
-                            channel.name()
+                            msg.get_name(),
+                            this_channel_name,
                         ));
                     }
-
+                    let this_channel_name = this_channel_name.map(|s| s.to_string());
```

Comment:
```
warning: cannot borrow `*self` as mutable because it is also borrowed as immutable
   --> mumd/src/state.rs:148:21
    |
138 |                 let this_channel_name = self.server().unwrap().channels().get(&this_channel).map(|c| c.name());
    |                                         ---- immutable borrow occurs here
...
148 |                     self.push_event(MumbleEventKind::UserConnected(msg.get_name().to_string(), this_channel_name.map(|s| s.to_string())));
    |                     ^^^^ mutable borrow occurs here                                            ----------------- immutable borrow later used here
    |
    = note: `#[warn(mutable_borrow_reservation_conflict)]` on by default
    = warning: this borrowing pattern was not meant to be accepted, and may become a hard error in the future
    = note: for more information, see issue #59159 <https://github.com/rust-lang/rust/issues/59159>
```

Reply:
Ah, maybe you meant the other way.